File: STRIVER/video/vid-5/pattern12/p12.cpp
// 1 - - - - - - 1
// 1 2 - - - - 2 1
// 1 2 3 - - 3 2 1
// 1 2 3 4 4 3 2 1

#include <iostream>
using namespace std;

int main()
{
    int row = 4;

    for (int i = 1; i <= row; i++)
    {

        // layer 1
        for (int j = 1; j <= i; j++)
        {
            cout << j;
        }

        // layer 2
        for (int j = 1; j <= 2 * (row - i); j++)
        {
            cout << " ";
        }

        // layer 3
        for (int j = i; j >= 1; j--)
        {
            cout << j ;
        }
        cout << endl;
    }

    return 0;
}

File: STRIVER/video/vid-5 (Patterns)/pattern12/p12.py
# 1 - - - - - - 1
# 1 2 - - - - 2 1
# 1 2 3 - - 3 2 1 
# 1 2 3 4 4 3 2 1


def p12(row:int)->None:
    for i in range(1,row+1):
    # layer 1
        for j in range(1,i+1):
            print(j,end="")
    # layer 2
        for j in range(1,2 *(row - i)+1):
            print(" ",end="")
    # layer 3
        for j in range(i, 0,-1):
            print(j,end="")
        print("")

row = 4
p12(row)